backend/src/main/java/com/bakdata/conquery/models/events/parser/ColumnValues.java

```diff
@@ -0,0 +1,44 @@
+package com.bakdata.conquery.models.events.parser;
+
+import java.util.BitSet;
+
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * per Column Store to encode null in auxiliary bitset, allowing primitive storage.
+ */
+@SuppressWarnings("Unchecked")
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public abstract class ColumnValues<T> {
+
+	private final T nullValue;
+	private final BitSet nulls = new BitSet();
+	private int size = 0;
+
+	public boolean isNull(int event) {
+		return nulls.get(event);
+	}
+
+	public abstract T get(int event);
+
+	public int add(T value) {
+		int event = size++;
+
+		if (value == null) {
+			nulls.set(event);
+			write(event, nullValue);
```

Comment:
Warum brauchst du noch einen Marker für Nullwerte in den den eigentlichen Daten?